contracts/CollateralRegistry.sol

```diff
@@ -22,7 +22,13 @@ contract CollateralRegistry is Auth {
 
     address[] _collaterals;
     
-    constructor(address _vaultParameters) Auth(_vaultParameters) {}
+    constructor(address _vaultParameters, address[] memory assets) Auth(_vaultParameters) {
+        for (uint i = 0; i < assets.length; i++) {
+            _collaterals.push(assets[i]);
+            collateralId[assets[i]] = i;
```

Comment:
Strictly speaking, a duplicate asset check should be added to ensure consistence.